src/functional/iterator/embedded.py

```diff
@@ -882,34 +924,38 @@ def get_ordered_indices(
                 res.append(elem[sparse_position_tracker[axis.value]])
                 sparse_position_tracker[axis.value] += 1
             else:
-                assert isinstance(elem, (int, np.integer, slice))
+                assert isinstance(elem, (int, np.integer, slice, range))
                 res.append(elem)
     return tuple(res)
 
 
-def _tupsum(a, b):
-    def combine_slice(s, t):
-        is_slice = False
-        if isinstance(s, slice):
-            is_slice = True
-            first = 0 if s.start is None else s.start
-            assert s.step is None
-            assert s.stop is None
-        else:
-            assert isinstance(s, numbers.Integral)
-            first = s
-        if isinstance(t, slice):
-            is_slice = True
-            second = 0 if t.start is None else t.start
-            assert t.step is None
-            assert t.stop is None
-        else:
-            assert isinstance(t, numbers.Integral)
-            second = t
-        start = first + second
-        return slice(start, None) if is_slice else start
+def _shift_range(
+    range_or_index: range | numbers.Integral, offset: numbers.Integral
```

Comment:
Does this really typechecks now? It should not work. The whole pyramid of numeric types in the standard library is older than `Protocols` and thus it is based on ABCs, which can only be typechecked dynamically (at run-type) but not statically. For example, this piece of code breaks in mypy:
```
a: numbers.Integral = int(3)
```
There are some complicated partial workarounds, but I think the only simple and stable solution for now is to just use `int` in the type annotations.

Reply:
replaced by IntIndex